static/src/javascripts/projects/facia/modules/ui/container-show-more.js

```diff
@@ -47,32 +60,105 @@ define([
              * DOM nodes.
              */
-            $container.toggleClass(className, button.state === 'displayed');
-            button.state = (button.state === 'hidden') ? 'displayed' : 'hidden';
-            setButtonState(button, button.state);
+            setButtonState(button, (button.state === STATE_HIDDEN) ? STATE_DISPLAYED : STATE_HIDDEN);
             updatePref(button.id, button.state);
         });
     }
 
-    function renderToDom($container, button) {
+    function renderToDom(button) {
         fastdom.write(function () {
-            $container.addClass(className)
+            button.$container.addClass(className)
                 .removeClass('js-container--fc-show-more')
-                .toggleClass(className, button.state === 'hidden');
+                .toggleClass(className, button.state === STATE_HIDDEN);
             // Initialise state, as it might be different from what was rendered server side based on localstorage prefs
             setButtonState(button, button.state);
         });
     }
 
+    function loadShowMore(pageId, containerId) {
+        return ajax({
+            url: '/' + pageId + '/show-more/' + containerId + '.json',
+            crossOrigin: true
+        });
+    }
+
+    function dedupShowMore($container, html) {
+        var seenArticles = itemsByArticleId($container),
+            $html = bonzo.create(html);
+
+        $(ITEM_SELECTOR, $html).each(function (article) {
+            var $article = bonzo(article);
+            if ($article.attr(ARTICLE_ID_ATTRIBUTE) in seenArticles) {
+                $article.remove();
+            }
+        });
+
+        return $html;
+    }
+
+    function loadShowMoreForContainer(button) {
+        fastdom.write(function () {
+            setButtonState(button, STATE_LOADING);
+        });
+
+        loadShowMore(config.page.pageId, button.id).then(function (response) {
+            var dedupedShowMore = dedupShowMore(button.$container, response.html);
+            fastdom.write(function () {
+                button.$placeholder.replaceWith(dedupedShowMore);
+                setButtonState(button, STATE_DISPLAYED);
+                updatePref(button.id, button.state);
+            });
+            button.isLoaded = true;
+        }, function () {
+            fastdom.write(function () {
+                setButtonState(button, STATE_HIDDEN);
+            });
+
+            showErrorMessage(button);
+        });
+    }
+
+    function hideErrorMessage($errorMessage) {
+        fastdom.write(function () {
+            $errorMessage.addClass('show-more__error-message--invisible');
+        });
+    }
+
+    function showErrorMessage(button) {
+        if (button.$errorMessage) {
+            button.$errorMessage.remove();
+        }
+
+        button.$errorMessage = bonzo(bonzo.create(
+            '<div class="show-more__error-message">' +
+                'Sorry, failed to load more stories. Please try again.' +
```

Comment:
shoudl we write the error message somewhere (sentry?) or is this already done?  I'm just imagining what happens when a user reports this error and we're asking them to use developer tools to find the failed request.